provider/internal/index.go

```diff
@@ -194,12 +197,12 @@ func (i *Index) Read(
 	provider.GetLogger(ctx).Debug("reading index with tag " + input.Tag)
 
 	digest, err := cli.ManifestInspect(ctx, input.Tag)
-	if err != nil && strings.Contains(err.Error(), "No such manifest:") && input.isPushed() {
+	if errors.Is(err, errs.ErrNotFound) {
 		// A remote tag was expected but isn't there -- delete the resource.
-		return "", input, state, err
+		return "", input, state, nil
```

Comment:
Based on the comment, the intent was to delete the resource here but returning a non-nil error would fail.